### PR TITLE
feat: support export otel telemetry directly over http(s)

### PR DESCRIPTION
**Pull request by markphelps**
Re: https://twitter.com/flipt_io/status/1705669178567364788

CleanShot 2023-09-23 at 15 32 24

TODO:

- [x] Tests
- [x] Test/support HTTPS
- [ ] Documentation/examples

## Testing

Tested with setting up hyperdx which only handles HTTP for otel-collection: https://www.hyperdx.io/docs/install/opentelemetry
Tested both using HTTP and HTTPS:

```
otlp:
    endpoint: "http://localhost:4318"
```

```
otlp:
    endpoint: "https://98f4-136-54-97-144.ngrok-free.app"
```

### CODE REVIEW

File: internal/cmd/grpc.go [hunk 1]
```diff
@@ -494,6 +479,69 @@ func (s *GRPCServer) onShutdown(fn errFunc) {
...
+			if err != nil {
+				traceExpErr = fmt.Errorf("parsing otlp endpoint: %w", err)
+				return
+			}
+
+			var client otlptrace.Client
+			switch u.Scheme {
+			case "http", "https":
+				client = otlptracehttp.NewClient(
+					otlptracehttp.WithEndpoint(u.Host+u.Path),
```
Comment (yquansah) [inline internal/cmd/grpc.go:510]: Are we able to just do something like this?

```suggestion
					otlptracehttp.WithEndpoint(u.String()),
```

File: internal/cmd/grpc.go [hunk 2]
```diff
@@ -494,6 +479,69 @@ func (s *GRPCServer) onShutdown(fn errFunc) {
...
+					otlptracehttp.WithHeaders(cfg.Tracing.OTLP.Headers),
+				)
+			case "grpc":
+				// TODO: support additional configuration options
+				client = otlptracegrpc.NewClient(
+					otlptracegrpc.WithEndpoint(u.Host+u.Path),
```
Comment (yquansah) [inline internal/cmd/grpc.go:516]: Same comment here.

File: internal/cmd/grpc_test.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+			if tt.wantErr != nil {
+				assert.EqualError(t, err, tt.wantErr.Error())
+				return
+			}
+			assert.NoError(t, err)
+			assert.NotNil(t, exp)
+			assert.NotNil(t, expFunc)
+		})
+	}
+}
```
Comment (yquansah) [inline internal/cmd/grpc_test.go:112]: 👌 Nice!

File: internal/cmd/grpc.go (hunk 1 above)
Comment (markphelps) [reply internal/cmd/grpc.go:510]: no because `otlptrace` expects the endpoint without the scheme. `u.String()` would return the scheme (`http/https/grpc`)

Comment (yquansah) [reply internal/cmd/grpc.go:510]: @markphelps Got it!

**Review by yquansah — APPROVED** [review]

**Pull request merged**